src/assets/pages/FightingArenaPage/FightingArenaPage.tsx

```diff
@@ -82,42 +89,59 @@ function FightingArenaPage() {
   function handleAttackClick() {
     setFightStatus(FightStatus.ATTACKING);
     const newDefenderHP = handleAttack({ selectedPokemon, opponentPokemon });
-    if (newDefenderHP) {
+    if (typeof newDefenderHP !== "undefined") {
       setOpponentPokemon((prev) => ({ ...prev, currentHP: newDefenderHP }));
     }
     setTimeout(() => {
-      if (newDefenderHP && newDefenderHP <= 0) {
+      if (newDefenderHP <= 0) {
         setFightStatus(FightStatus.WON);
+        return;
       }
       const newPlayerHP = handleAttack({ opponentPokemon, selectedPokemon });
-      if (newPlayerHP) {
+      if (typeof newPlayerHP !== "undefined") {
         setSelectedPokemon((prev) => ({ ...prev, currentHP: newPlayerHP }));
       }
       setTimeout(() => {
-        if (newPlayerHP && newPlayerHP <= 0) {
+        if (newPlayerHP <= 0) {
           setFightStatus(FightStatus.LOST);
         }
-       
-          setFightStatus((prev) => (prev === FightStatus.ATTACKING ? FightStatus.IN_FIGHT : prev));
+
+        setFightStatus((prev) =>
+          prev === FightStatus.ATTACKING ? FightStatus.IN_FIGHT : prev
+        );
       }, TIME_BETWEEN_TURNS);
     }, TIME_BETWEEN_TURNS);
   }
   function handleCatchClick() {
     setFightStatus(FightStatus.CATCHING);
     const catchStatus = attemptCatch(opponentPokemon.currentHP);
     setCatchAttemptsLeft((prev) => prev - 1);
-  
-    setTimeout(() => {
-     if(catchStatus){
-      setFightStatus(FightStatus.CATCHED);
-     }else if (catchAttemptsLeft <= 0) {
-      setFightStatus(FightStatus.LOST);
-    } else {
-      setFightStatus(FightStatus.IN_FIGHT);
-    }
+
+    setTimeout(async () => {
+      if (catchStatus) {
+        setFightStatus(FightStatus.CATCHED);
+        const caughtPokemon = await catchPokemon(
+          opponentPokemon.pokemon?.id || ""
+        );
+        setTimeout(() => {
+          setFightStatus(FightStatus.WON);
+        }, TIME_BETWEEN_TURNS);
+      } else if (catchAttemptsLeft - 1 <= 0) {
```

Comment:
```suggestion
      } else if (catchAttemptsLeft <=1)  {
```

i think it the same check and its more readable